src/Compilers/VisualBasic/Portable/Symbols/Symbol.vb

```diff
@@ -604,7 +604,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                          SymbolKind.Event,
                          SymbolKind.Parameter,
                          SymbolKind.TypeParameter,
-                         SymbolKind.ErrorType
+                         SymbolKind.ErrorType,
+                         SymbolKind.Preprocessing
```

Comment:
It doesn't look like this change is covered by a targeted test under Compilers.

Reply:
Is this change even meaningful? It doesn't look like similar change is made on C# side.

Reply:
This is required by the FAR engine to detect whether the symbol usage can be discovered by searching for documents that contain identifiers with the name of the symbol. The C# `PreprocessingSymbol` implements `CanBeReferencedByName` on its own, so that's why we did not have to make a change there. The VB version however was not handling it and so the case was added in the base implementation without checking the identifier's validity.

This property is barely tested targeted across the solution. I don't know what would be a good test for this property other than a trivial "we have a preprocessing symbol and the property returns true". Though not keeping this change breaks the FAR for pp symbols.

Reply:
> I don't know what would be a good test for this property other than a trivial "we have a preprocessing symbol and the property returns true". 

Having a simple test like that for a public API behavior should be sufficient. I think. Consider adding a test for C# behavior as well, if we don't already have one.